trainers/dataloader.py

```diff
@@ -66,11 +68,12 @@ def __getitem__(self, idx):
         
         Note: This works for BytePooling and StandardDataloader, but not for ConversationalDataloader and NextTokenMLMDataloader
         '''
+        idx = random.randint(0, len(self.data) - self.context_window) # lol
```

Comment:
think we can append a comment here that goes something like this? 

"## implementing our own random sampling here to avoid issues with Pytorch's DataLoader". So that no one accidentally thinks this is redundant and deletes it.